app/scripts/controllers/preferences.test.js

```diff
@@ -109,6 +113,24 @@ describe('preferences controller', () => {
     });
   });
 
+  describe('onAccountRemoved', () => {
+    it('should call removeAddress method with the correct address when the `accountRemoved` event is emitted.', () => {
+      const mockAddress = '0x1234567890abcdef';
+      const removeAddressMock = jest.fn();
+      preferencesController.removeAddress = removeAddressMock;
+      mockEventEmitter.emit('accountRemoved', mockAddress);
+      expect(removeAddressMock).toHaveBeenCalledWith(mockAddress);
+    });
+
+    it('should never call removeAddress method when `accountRemoved` is emitted with the wrong event name', () => {
+      const mockAddress = '0x1234567890abcdef';
+      const removeAddressMock = jest.fn();
+      preferencesController.removeAddress = removeAddressMock;
+      mockEventEmitter.emit('wrongAccountRemoved', mockAddress);
+      expect(removeAddressMock).not.toHaveBeenCalled();
```

Comment:
I think that `PreferencesController` should not know which exact name the event has, as it is passed to it during construction. 

Perhaps it would be more correct to test that when calling `onAccountRemoved` with a non-existing address the identities in the state remain the same